examples/order_manager/order_service/order_service.bal

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerinax/kafka;
+import ballerina/log;
+import order_service.types;
+import ballerina/random;
+
+configurable string TOPIC = "orders";
+configurable int LISTENER_PORT = 9090;
+
+// Creates a Kafka producer with default configurations
+kafka:Producer kafkaProducer = check new (kafka:DEFAULT_URL);
+
+@http:ServiceConfig {
+    auth: [
+        {
+            fileUserStoreConfig: {},
+            scopes: ["customer"]
+        }
+    ]
+}
+service /kafka on new http:Listener(LISTENER_PORT) {
+
+    // Listens to new order requests and publishes the order to Kafka topic
+    resource function get publish(string message, string status) returns string|error? {
+        // Converts the request parameter status to type PaymentStatus
+        types:PaymentStatus paymentStatus = <types:PaymentStatus> status;
```

Comment:
Shall we use `value:ensureType` rather than casting here ?

Reply:
Changed in https://github.com/ballerina-platform/module-ballerinax-kafka/pull/278/commits/032865c1873c336a22587e040c46c3e8188a3b49